core/templates/dev/head/pages/topic_editor/main_editor/TopicEditorTabDirective.js

```diff
@@ -0,0 +1,67 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Controller for the main topic editor.
+ */
+
+oppia.directive('topicMainEditor', [
+  'UrlInterpolationService', function(UrlInterpolationService) {
+    return {
+      restrict: 'E',
+      scope: {},
+      templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+        '/pages/topic_editor/main_editor/topic_editor_tab_directive.html'),
+      controller: [
+        '$scope', 'TopicEditorStateService', 'TopicUpdateService',
+        'EVENT_TOPIC_INITIALIZED', 'EVENT_TOPIC_REINITIALIZED',
+        function(
+            $scope, TopicEditorStateService, TopicUpdateService,
+            EVENT_TOPIC_INITIALIZED, EVENT_TOPIC_REINITIALIZED) {
+          $scope.topic = TopicEditorStateService.getTopic();
+          $scope.topicRights = TopicEditorStateService.getTopicRights();
+          $scope.canEditName = GLOBALS.isAdmin;
```

Comment:
Don't do this, refer to $scope.topicRights and make sure that that has sufficient info to determine this.

Reply:
In the docs, it was written that topic managers also won't be allowed to change the name, so shall I send an isAdmin also in topic_rights?

Reply:
Done.